modules/javafx.graphics/src/main/native-prism-d3d/D3DLight.cc

```diff
@@ -56,3 +60,7 @@ void D3DLight::setPosition(float x, float y, float z) {
     position[1] = y;
     position[2] = z;
 }
+
+/*void D3DLight::setRange(float r) {
```

Comment:
I forgot to mention this point. These setter functions for color and position (and range) are never used since whenever there is a change in the java side the whole array and lights are recreated instead of being adjusted for the change. I'm not familiar enough with the memory model of JNI, but it seems expensive to re-render everything on every change (I think that the mesh is also recreated every time in the native code). Is this the way it's supposed to work?

Reply:
If the lights array is being recreated on a change, there might be a small savings to be had to reuse the existing array (it would need to be updated only during the sync while the renderer is idle).

If the mesh is always being recreated, there may be an opportunity for an even bigger savings, presuming that only the mesh data changed (and not the index values in the faces array).

This could be a future optimization, but we would need something to quantify the possible savings.